Current issues with direct Nearby uploads

**Summary:** 

I thought we should probably keep a list of issues we are experiencing with the directNearbyUploadsNew feature branch, so we don't miss anything out before releasing to production. Please feel free to edit this post to add issues or detail for issues.

**Steps to reproduce:** 

Build from `2.7.x-release` and test the Nearby feature

- [x] 1. Memory leaks PR #1390 
- [x] 2. Crash when loading map (by the time I finished typing this out, I had lost the logcat. Sorry. :/ Will post when I encounter it again).
- [x] 3. <s>Text wrap issues in bottom sheet (see SS below). I would recommend removing "Directions" text</s>
- [x] 4. Fix "?" description bug in map (see SS below). This has already been fixed in list, we just need to copy the fix over to map. I should be able to do it after I'm done with my category PR.
- [x] 5. <s> Crash on orientation change</s> crashes when list view is open and orinetation changed. PR: #1385 
- [x] 6.  Bug: if you click one of nearby markers and make information bottom sheet visible, then go to settings and change theme, app will crash when you re-start nearby activity. Log is at below comment PR #1391
- [x] 7. <s> Action bar is hidden</s>
- [x] 8. Fix current location point visibility when bottom sheet is expanded, and re-center button is clicked. PR: #1381
- [x] 9. Fix transparent view is top of menu bar - PR is ready waiting for review #1378

![screenshot_20180318-194400](https://user-images.githubusercontent.com/3611199/37564649-ece24dc0-2ae5-11e8-817d-974f7d32a16d.png)

```
 FATAL EXCEPTION: RxCachedThreadScheduler-1
                                                                           Process: fr.free.nrw.commons.debug, PID: 19653
                                                                           io.reactivex.exceptions.UndeliverableException: java.lang.IndexOutOfBoundsException
                                                                               at io.reactivex.plugins.RxJavaPlugins.onError(RxJavaPlugins.java:349)
                                                                               at io.reactivex.internal.operators.observable.ObservableFromCallable.subscribeActual(ObservableFromCallable.java:48)
                                                                               at io.reactivex.Observable.subscribe(Observable.java:10901)
                                                                               at io.reactivex.internal.operators.observable.ObservableSubscribeOn$SubscribeTask.run(ObservableSubscribeOn.java:96)
                                                                               at io.reactivex.Scheduler$DisposeTask.run(Scheduler.java:452)
                                                                               at io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:61)
                                                                               at io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:52)
                                                                               at java.util.concurrent.FutureTask.run(FutureTask.java:237)
                                                                               at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:152)
                                                                               at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:265)
                                                                               at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
                                                                               at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
                                                                               at java.lang.Thread.run(Thread.java:818)
                                                                            Caused by: java.lang.IndexOutOfBoundsException
                                                                               at java.util.Collections$EmptyList.get(Collections.java:102)
                                                                               at fr.free.nrw.commons.nearby.NearbyController.loadAttractionsFromLocation(NearbyController.java:57)
                                                                               at fr.free.nrw.commons.nearby.NearbyActivity.lambda$refreshView$6$NearbyActivity(NearbyActivity.java:317)
                                                                               at fr.free.nrw.commons.nearby.NearbyActivity$$Lambda$6.call(Unknown Source)
                                                                               at io.reactivex.internal.operators.observable.ObservableFromCallable.subscribeActual(ObservableFromCallable.java:42)
                                                                               at io.reactivex.Observable.subscribe(Observable.java:10901) 
                                                                               at io.reactivex.internal.operators.observable.ObservableSubscribeOn$SubscribeTask.run(ObservableSubscribeOn.java:96) 
                                                                               at io.reactivex.Scheduler$DisposeTask.run(Scheduler.java:452) 
                                                                               at io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:61) 
                                                                               at io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:52) 
                                                                               at java.util.concurrent.FutureTask.run(FutureTask.java:237) 
                                                                               at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:152) 
                                                                               at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:265) 
                                                                               at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112) 
                                                                               at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587) 
                                                                               at java.lang.Thread.run(Thread.java:818) 
```

I still get this crash when orientation changes when the options on nearby list is enabled

```
03-21 16:59:08.519 24135-24135/fr.free.nrw.commons.debug E/AndroidRuntime: FATAL EXCEPTION: main
                                                                           Process: fr.free.nrw.commons.debug, PID: 24135
                                                                           io.reactivex.exceptions.OnErrorNotImplementedException
                                                                               at io.reactivex.internal.functions.Functions$OnErrorMissingConsumer.accept(Functions.java:704)
                                                                               at io.reactivex.internal.functions.Functions$OnErrorMissingConsumer.accept(Functions.java:701)
                                                                               at io.reactivex.internal.observers.LambdaObserver.onError(LambdaObserver.java:74)
                                                                               at io.reactivex.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.checkTerminated(ObservableObserveOn.java:276)
                                                                               at io.reactivex.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.drainNormal(ObservableObserveOn.java:172)
                                                                               at io.reactivex.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.run(ObservableObserveOn.java:252)
                                                                               at io.reactivex.android.schedulers.HandlerScheduler$ScheduledRunnable.run(HandlerScheduler.java:109)
                                                                               at android.os.Handler.handleCallback(Handler.java:739)
                                                                               at android.os.Handler.dispatchMessage(Handler.java:95)
                                                                               at android.os.Looper.loop(Looper.java:135)
                                                                               at android.app.ActivityThread.main(ActivityThread.java:5343)
                                                                               at java.lang.reflect.Method.invoke(Native Method)
                                                                               at java.lang.reflect.Method.invoke(Method.java:372)
                                                                               at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:905)
                                                                               at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:700)
                                                                            Caused by: java.lang.IndexOutOfBoundsException
                                                                               at java.util.Collections$EmptyList.get(Collections.java:102)
                                                                               at fr.free.nrw.commons.nearby.NearbyController.loadAttractionsFromLocation(NearbyController.java:57)
                                                                               at fr.free.nrw.commons.nearby.NearbyActivity.lambda$refreshView$6$NearbyActivity(NearbyActivity.java:317)
                                                                               at fr.free.nrw.commons.nearby.NearbyActivity$$Lambda$6.call(Unknown Source)
                                                                               at io.reactivex.internal.operators.observable.ObservableFromCallable.subscribeActual(ObservableFromCallable.java:42)
                                                                               at io.reactivex.Observable.subscribe(Observable.java:10901)
                                                                               at io.reactivex.internal.operators.observable.ObservableSubscribeOn$SubscribeTask.run(ObservableSubscribeOn.java:96)
                                                                               at io.reactivex.Scheduler$DisposeTask.run(Scheduler.java:452)
                                                                               at io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:61)
                                                                               at io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:52)
                                                                               at java.util.concurrent.FutureTask.run(FutureTask.java:237)
                                                                               at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:152)
                                                                               at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:265)
                                                                               at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
                                                                               at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
                                                                               at java.lang.Thread.run(Thread.java:818)

```

I get this crash when internet is  unavailable

I still get the below crash occasionally when I change orientation in Nearby, or sometimes even if I go to another activity and then return (e.g. uploading, or wikipedia, etc). It doesn't happen reliably, though.

```
03-26 18:37:09.683: E/AndroidRuntime(2741): FATAL EXCEPTION: main
03-26 18:37:09.683: E/AndroidRuntime(2741): Process: fr.free.nrw.commons.debug, PID: 2741
03-26 18:37:09.683: E/AndroidRuntime(2741): io.reactivex.exceptions.OnErrorNotImplementedException: Index: 0
03-26 18:37:09.683: E/AndroidRuntime(2741): 	at io.reactivex.internal.functions.Functions$OnErrorMissingConsumer.accept(Functions.java:704)
03-26 18:37:09.683: E/AndroidRuntime(2741): 	at io.reactivex.internal.functions.Functions$OnErrorMissingConsumer.accept(Functions.java:701)
03-26 18:37:09.683: E/AndroidRuntime(2741): 	at io.reactivex.internal.observers.LambdaObserver.onError(LambdaObserver.java:74)
03-26 18:37:09.683: E/AndroidRuntime(2741): 	at io.reactivex.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.checkTerminated(ObservableObserveOn.java:276)
03-26 18:37:09.683: E/AndroidRuntime(2741): 	at io.reactivex.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.drainNormal(ObservableObserveOn.java:172)
03-26 18:37:09.683: E/AndroidRuntime(2741): 	at io.reactivex.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.run(ObservableObserveOn.java:252)
03-26 18:37:09.683: E/AndroidRuntime(2741): 	at io.reactivex.android.schedulers.HandlerScheduler$ScheduledRunnable.run(HandlerScheduler.java:109)
03-26 18:37:09.683: E/AndroidRuntime(2741): 	at android.os.Handler.handleCallback(Handler.java:751)
03-26 18:37:09.683: E/AndroidRuntime(2741): 	at android.os.Handler.dispatchMessage(Handler.java:95)
03-26 18:37:09.683: E/AndroidRuntime(2741): 	at android.os.Looper.loop(Looper.java:154)
03-26 18:37:09.683: E/AndroidRuntime(2741): 	at android.app.ActivityThread.main(ActivityThread.java:6692)
03-26 18:37:09.683: E/AndroidRuntime(2741): 	at java.lang.reflect.Method.invoke(Native Method)
03-26 18:37:09.683: E/AndroidRuntime(2741): 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1468)
03-26 18:37:09.683: E/AndroidRuntime(2741): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1358)
03-26 18:37:09.683: E/AndroidRuntime(2741): Caused by: java.lang.IndexOutOfBoundsException: Index: 0
03-26 18:37:09.683: E/AndroidRuntime(2741): 	at java.util.Collections$EmptyList.get(Collections.java:3690)
03-26 18:37:09.683: E/AndroidRuntime(2741): 	at fr.free.nrw.commons.nearby.NearbyController.loadAttractionsFromLocation(NearbyController.java:57)
03-26 18:37:09.683: E/AndroidRuntime(2741): 	at fr.free.nrw.commons.nearby.NearbyActivity.lambda$refreshView$6$NearbyActivity(NearbyActivity.java:351)
03-26 18:37:09.683: E/AndroidRuntime(2741): 	at fr.free.nrw.commons.nearby.NearbyActivity$$Lambda$6.call(Unknown Source)
03-26 18:37:09.683: E/AndroidRuntime(2741): 	at io.reactivex.internal.operators.observable.ObservableFromCallable.subscribeActual(ObservableFromCallable.java:42)
03-26 18:37:09.683: E/AndroidRuntime(2741): 	at io.reactivex.Observable.subscribe(Observable.java:10901)
03-26 18:37:09.683: E/AndroidRuntime(2741): 	at io.reactivex.internal.operators.observable.ObservableSubscribeOn$SubscribeTask.run(ObservableSubscribeOn.java:96)
03-26 18:37:09.683: E/AndroidRuntime(2741): 	at io.reactivex.Scheduler$DisposeTask.run(Scheduler.java:452)
03-26 18:37:09.683: E/AndroidRuntime(2741): 	at io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:61)
03-26 18:37:09.683: E/AndroidRuntime(2741): 	at io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:52)
03-26 18:37:09.683: E/AndroidRuntime(2741): 	at java.util.concurrent.FutureTask.run(FutureTask.java:237)
03-26 18:37:09.683: E/AndroidRuntime(2741): 	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:272)
03-26 18:37:09.683: E/AndroidRuntime(2741): 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
03-26 18:37:09.683: E/AndroidRuntime(2741): 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
03-26 18:37:09.683: E/AndroidRuntime(2741): 	at java.lang.Thread.run(Thread.java:762)

```

--- a/app/src/main/res/layout/activity_nearby.xml
+++ b/app/src/main/res/layout/activity_nearby.xml
@@ -5,7 +5,7 @@
     android:layout_width="match_parent"
     android:layout_height="match_parent">
 
-    <android.support.design.widget.CoordinatorLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    <android.support.design.widget.CoordinatorLayout
         xmlns:app="http://schemas.android.com/apk/res-auto"
         android:id="@+id/coordinator_layout"
         android:layout_width="match_parent"
@@ -39,32 +39,13 @@
                     android:layout_width="match_parent"
                     android:layout_height="match_parent" />
             </LinearLayout>
-            <android.support.design.widget.FloatingActionButton
-                android:id="@+id/fab_list"
-                android:layout_width="wrap_content"
-                android:layout_height="wrap_content"
-                app:fabSize="mini"
-                android:layout_alignBottom="@id/toolbar"
-                android:layout_alignParentRight="true"
-                android:layout_marginRight="28dp"
-                android:layout_marginBottom="-96dp"
-                android:visibility="invisible"
-                app:elevation="6dp"
-                app:pressedTranslationZ="12dp"
-                app:backgroundTint="@color/button_blue"
-                android:clickable="true"
-                app:srcCompat="@drawable/ic_list_white_24dp"
-                android:scaleType="center"
-                />
 
             <View
                 android:id="@+id/transparentView"
                 android:layout_width="match_parent"
                 android:layout_height="match_parent"
                 android:layout_alignParentLeft="true"
-                android:layout_alignParentStart="true"
-                android:layout_alignParentTop="true"
-                android:layout_marginTop="18dp"
+                android:layout_below="@id/toolbar"
                 android:background="#aa969696"
                 android:elevation="6dp">
 

